### PR TITLE
Dev main script 20180918

### PR DESCRIPTION
**Pull request by JamieGainer**
This is merging the commits I had up to Wednesday evening into the main branch.  I will start the main .py file in the autarchy directory to make the architecture clear (in a new branch).

### CODE REVIEW

File: meta_model_building/get_opt_params.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+	'test_size': 0.25,
+	'generations': 5,
+	'population_size': 20,
+	'verbosity': 2,
+	'time': time.time()
+}
+
+housing = load_boston()
+X_train, X_test, y_train, y_test = train_test_split(housing.data, housing.target,
+                                                    train_size=run_param['train_size'],
```
Comment (agude) [inline meta_model_building/get_opt_params.py]: You're using inconsistent indenting.

File: meta_model_building/get_opt_params.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+}
+
+housing = load_boston()
+X_train, X_test, y_train, y_test = train_test_split(housing.data, housing.target,
+                                                    train_size=run_param['train_size'], 
+                                                    test_size=run_param['test_size'],
+						    random_state = run_param['random_state_train'])
+
+tpot = TPOTRegressor(generations=run_param['generations'], population_size=run_param['population_size'], 
+					 verbosity=run_param['verbosity'], random_state = run_param['random_state_tpot'])
```
Comment (agude) [inline meta_model_building/get_opt_params.py]: NEVER mix tabs and spaces! Python3 actually makes it an error to do so. https://www.python.org/dev/peps/pep-0008/?#tabs-or-spaces

File: meta_model_building/get_opt_params.py [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+	'train_size': 0.75,
+	'test_size': 0.25,
+	'generations': 5,
+	'population_size': 20,
+	'verbosity': 2,
+	'time': time.time()
+}
+
+housing = load_boston()
+X_train, X_test, y_train, y_test = train_test_split(housing.data, housing.target,
```
Comment (agude) [inline meta_model_building/get_opt_params.py]: In general I'm not a fan of mixing upper and lower case like this.

File: meta_model_building/get_opt_params.py [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+                                                    train_size=run_param['train_size'], 
+                                                    test_size=run_param['test_size'],
+						    random_state = run_param['random_state_train'])
+
+tpot = TPOTRegressor(generations=run_param['generations'], population_size=run_param['population_size'],
```
Comment (agude) [inline meta_model_building/get_opt_params.py]: You can probably do `TPOTRegressor(**run_param)` which will unpack the keywords into the right keyword arguments. https://stackoverflow.com/a/334666

File: meta_model_building/get_opt_params.py [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+X_train, X_test, y_train, y_test = train_test_split(housing.data, housing.target,
+                                                    train_size=run_param['train_size'], 
+                                                    test_size=run_param['test_size'],
+						    random_state = run_param['random_state_train'])
+
+tpot = TPOTRegressor(generations=run_param['generations'], population_size=run_param['population_size'], 
+					 verbosity=run_param['verbosity'], random_state = run_param['random_state_tpot'])
+
+tpot.fit(X_train, y_train)
+print(tpot.score(X_test, y_test))
```
Comment (agude) [inline meta_model_building/get_opt_params.py]: Prefer `logging` to `print`. I use print a lot, but logging is the "professional" way. ;-)

File: meta_model_building/get_opt_params.py [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+tpot.export(run_param['export_python_code_filename'])
+
+# Prepare metadata dictionary for pickling
+pickle_dict = tpot.evaluated_individuals_
```
Comment (agude) [inline meta_model_building/get_opt_params.py]: A trailing `_` generally means you shouldn't access that directly; is there some accessor function instead?

File: meta_model_building/get_opt_params.py (hunk 2 above)
Comment (JamieGainer) [reply meta_model_building/get_opt_params.py]: Fixed in first commit to dev_get-boston-data_20180923 branch.

File: meta_model_building/get_opt_params.py (hunk 3 above)
Comment (JamieGainer) [reply meta_model_building/get_opt_params.py]: Fixed in second commit to dev_get-boston-data_20180923

File: meta_model_building/get_opt_params.py (hunk 4 above)
Comment (JamieGainer) [reply meta_model_building/get_opt_params.py]: I've implemented this in the current dev branch.

File: meta_model_building/get_opt_params.py (hunk 6 above)
Comment (JamieGainer) [reply meta_model_building/get_opt_params.py]: It doesn't seem so.  In general a major problem that I have with TPOT is that it doesn't make it convenient (with accessor functions, etc.) to access intermediate data and/or assess algorithm progress.

File: meta_model_building/get_opt_params.py (hunk 1 above)
Comment (JamieGainer) [reply meta_model_building/get_opt_params.py]: I've PEP-8-ified the current version (in dev_get-boston-data_20180924-- I'll put in a pull request for this branch today).

**Pull request merged**